source/command-monitoring/tests/unified/find.yml

```diff
@@ -0,0 +1,290 @@
+    documents:
+      - { _id: 1, x: 11 }
+      - { _id: 2, x: 22 }
+      - { _id: 3, x: 33 }
+      - { _id: 4, x: 44 }
+      - { _id: 5, x: 55 }
+
+tests:
+  - description: "A successful find with no options"
+    operations:
+      - name: find
+        object: *collection
+        arguments:
+          filter: { _id: 1 }
+    expectEvents:
+      - client: *client
+        events:
+          - commandStartedEvent:
+              command:
+                find: *collectionName
+                filter: { _id: 1 }
+              commandName: find
+              databaseName: *databaseName
+          - commandSucceededEvent:
+              reply:
+                ok: 1
+                cursor:
+                  id: 0
+                  ns: *namespace
+                  firstBatch:
+                    -  { _id: 1, x: 11 }
+              commandName: find
+
+  - description: "A successful find with options"
+    operations:
+      - name: find
+        object: *collection
+        arguments:
+          filter: { _id: { $gt: 1 } }
+          sort: { _id: 1 }
+          skip: 2
+          comment: "test"
+          hint: { _id: 1 }
+          max: { _id: 6 }
+          maxTimeMS: 6000
+          min: { _id: 0 }
+          returnKey: false
+          showRecordId: false
+          readPreference:
+            mode: primaryPreferred
+    expectEvents:
+      - client: *client
+        events:
+          - commandStartedEvent:
+              command:
+                find: *collectionName
+                filter: { _id: { $gt: 1 } }
+                sort: { _id: 1 }
+                skip: 2
+                comment: "test"
+                hint: { _id: 1 }
+                max: { _id: 6 }
+                maxTimeMS: 6000
+                min: { _id: 0 }
+                returnKey: false
+                showRecordId: false
+              commandName: find
+              databaseName: *databaseName
+          - commandSucceededEvent:
+              reply:
+                ok: 1
+                cursor:
+                  id: 0
+                  ns: *namespace
+                  firstBatch:
+                    - { _id: 4, x: 44 }
+                    - { _id: 5, x: 55 }
+              commandName: find
+
+  - description: "A successful find with a getMore"
+    operations:
+      - name: find
+        object: *collection
+        arguments:
+          filter: { _id: { $gte: 1 }}
+          sort: { _id: 1 }
+          batchSize: 3
+    expectEvents:
+      - client: *client
+        events:
+          - commandStartedEvent:
+              command:
+                find: *collectionName
+                filter: { _id: { $gte: 1 }}
+                sort: { _id: 1 }
+                batchSize: 3
+              commandName: find
+              databaseName: *databaseName
+          - commandSucceededEvent:
+              reply:
+                ok: 1
+                cursor:
+                  id: { $$type: [ int, long ] }
```

Comment:
the legacy runner wants you to assert consistency between non-zero cursor ids within single tests, but I couldn't think of a way to do that in the unified format, and that also seems more like a test for cursor behavior than command monitoring behavior so it didn't seem too important to try to replicate.

Reply:
I concur. Asserting that `getMore` uses the same cursor ID value seems beyond the scope of this spec. When/if we get a cursor spec, that seems like something we can offload to a prose test. Granted, if drivers weren't behaving correctly in that regard a _lot_ of spec tests would start failing, so I'm also not too concerned about this.